torchtune/models/flamingo/_encoder.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional
+
+import torch
+
+from torch import nn, Tensor
+from torchtune.modules.model_fusion import register_fusion_module
+from torchtune.modules.transformer import _get_clones
+
+
+class FlamingoProjectionHead(nn.Module):
```

Comment:
nit: is Head the right term? I would think Head is something that attached on top of the hidden states of a transformer and not a full model. I thought we had referred to this as an Adapter?

Reply:
A projection heads have been around a long time and can vary a lot in architecture. The main point here is that it's learning a projection from the pre-trained encoder to the pretrained decoder.